opentreemap/treemap/views/tree.py

```diff
@@ -90,16 +90,57 @@ def search_tree_benefits(request, instance):
     }
 
     formatted = format_benefits(instance, benefits, basis, digits=0)
-    formatted['hide_summary'] = hide_summary
 
-    formatted['tree_count_label'] = (
-        'tree,' if basis['plot']['n_total'] == 1 else 'trees,')
-    formatted['plot_count_label'] = (
-        'planting site' if basis['plot']['n_plots'] == 1 else 'planting sites')
-    if instance.has_resources and 'resource' in basis:
-        formatted['plot_count_label'] += ','
+    n_trees = basis['plot']['n_total']
+    n_plots = basis['plot']['n_plots']
+    n_empty_plots = n_plots - n_trees
+    n_resources = 0
+
+    tree_count_label = ungettext('tree', 'trees', n_trees) + ','
+    empty_plot_count_label = ungettext(
+        'empty planting site', 'empty planting sites', n_empty_plots)
+    has_resources = instance.has_resources and 'resource' in basis
+    if has_resources:
+        n_resources = basis['resource']['n_total']
+        empty_plot_count_label += ','
+
+    context = {
+        'n_trees': n_trees,
+        'n_empty_plots': n_empty_plots,
+        'n_resources': n_resources,
+        'tree_count_label': tree_count_label,
+        'empty_plot_count_label': empty_plot_count_label,
+        'has_resources': has_resources,
+        'hide_summary': hide_summary,
+        'single_result': _single_result_context(instance, n_plots, n_resources,
+                                                filter)
+    }
+    context.update(formatted)
+    return context
+
 
-    return formatted
+def _single_result_context(instance, n_plots, n_resources, filter):
+    # If search found just one feature, return its id and location
+    if n_plots + n_resources != 1:
+        return None
+    else:
+        if n_plots == 1:
+            qs = filter.get_objects(Plot)
+        else:  # n_resources == 1
+            for Resource in instance.resource_classes:
+                qs = filter.get_objects(Resource)
+                if qs.count() == 1:
+                    break
+        if qs.count() != 1:
+            raise Exception('Failed to retrieve single resource')
```

Comment:
Is this unlikely enough that omitting this test and just letting it blow up on the next line where it indexes qs should be sufficient?